Tensile/Utilities/tensile_power.sh

```diff
@@ -0,0 +1,184 @@
+    if [ -z "${cmd}" ]; then
+        echo "${ATITOOL_PATH}/atitool -i=$card_id -ppdpmforce=gfx,${gfx_level} Failed"
+    fi
+    cmd=`${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id} -ppdpmforce=mclk,${mclk_level}`
+    cmd=`${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id} -ppdpmforce=pcie,1`
+    cmd=`${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id} -ppdpmforce=fclk,7`
+    cmd=`${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id} -ppdpmforce=soc,7`
+  elif [ "$dpm" -eq "on" ]; then
+    cmd=`${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id}  -ppdpmrestore`
+  fi 
+ cmd=`sleep 5s` # let clk force take effect
+ log_tag="dpmForce_gfx${gfx_clk}_mclk${mclk}_socclk${soc_clk}_${time_stamp}"
+
+ echo "#####################################################################"
+ echo "Start background process atitool for capturing power data....."
+ echo "#####################################################################"
+ cmd="${sudo_cmd} ${ATITOOL_PATH}/atitool -i=${card_id} -pmlogall -pmperiod=50 -pmstopcheck -pmoutput=${output_dir}/${log_tag}_pmlog.csv"
+ eval "${cmd}" &>/dev/null &disown
+ tensile_cmd="HIP_VISIBLE_DEVICES=${card_id} HSA_ENABLE_SDMA=1  python ./Tensile/Tensile.py ${yaml_filename} build_pmrun 2>&1 | tee ${output_dir}/${log_tag}.log"
+ eval "${tensile_cmd}" 
+}
+
+generate_report=0
+ARG_IN=""
+while [ 1 ] ; do
+  ARG_IN=$1
+  ARG_VAL=1
+  if [ "$1" = "-h" ] ; then
+    usage
+  elif [ "$1" = "-i" ] ; then
+    yaml_filename="$2"
+  elif [ "$1" = "-d" ] ; then
+    output_dir="$2"
+  elif [ "$1" = "-gclk" ] ; then
+    gfx_clk="$2"
+  elif [ "$1" = "-mclk" ] ; then
+    mclk="$2"
+  elif [ "$1" = "-socclk" ] ; then
+    soc_clk="$2"
+  elif [ "$1" = "--all" ] ; then
+    best_allsizes=0
+    ARG_VAL=0
+  elif [ "$1" = "--genrpt" ] ; then
+    generate_report=1
+    ARG_VAL=0
+  else
+    break
+  fi
+  shift
+  if [ "$ARG_VAL" = 1 ] ; then shift; fi
+done
+
+ARG_CK=`echo $ARG_IN | sed "s/^-.*$/-/"`
+if [ "$ARG_CK" = "-" ] ; then
+  fatal "Wrong option '$ARG_IN'"
+fi
+
+if [ -z "$yaml_filename" ] ; then
+  fatal "Need input file"
+fi
+
+if [ -z "$output_dir" ] ; then
+   output_dir="./tensile_pmlogs"
+fi
+
+## check ATITOOLPATH variable
+if [ -z "${ATITOOL_PATH}" ]; then
+    fatal "ATITOOL_PATH is not set"
+else
+    echo "ATITOOL_PATH is set: $ATITOOL_PATH"
+fi
+
+echo "##########################################################"
+echo "Creating output directory $output_dir......"
+echo "##########################################################"
+mkdir -p "${output_dir}"
+
+#for i in 4 do
+set_dpm_and_run "force" ${gfxclk_array[$gfx_clk]} ${mclk_array[$mclk]}
+#end
+
+##genrate report calling
+
+echo "##########################################################"
+echo "Generating consolidated output pmlog.csv ....."
+echo "##########################################################"
+if [ "${generate_report}" = 1 ]; then
+   cmd=`ruby ./Tensile/Utilities/gen_csv.rb  ${output_dir} ${best_allsizes}`
+fi
+
+exit 1
```

Comment:
what is this file for?

Reply:
script for generating power reports.. not sure how did that got into this PR.. I will remove the file..